OrbitClientData/include/OrbitClientData/ModuleData.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_MODULE_DATA_H_
+#define ORBIT_GL_MODULE_DATA_H_
+
+#include <cinttypes>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/str_format.h"
+#include "absl/synchronization/mutex.h"
+#include "capture_data.pb.h"
+#include "module.pb.h"
+#include "symbol.pb.h"
+
+// Represents information about module on the client
+class ModuleData final {
+ public:
+  explicit ModuleData(orbit_grpc_protos::ModuleInfo info)
+      : module_info_(std::move(info)), is_loaded_(false) {}
+
+  [[nodiscard]] const std::string& name() const { return module_info_.name(); }
+  [[nodiscard]] const std::string& file_path() const { return module_info_.file_path(); }
+  [[nodiscard]] uint64_t file_size() const { return module_info_.file_size(); }
+  [[nodiscard]] const std::string& build_id() const { return module_info_.build_id(); }
+  [[nodiscard]] uint64_t load_bias() const { return module_info_.load_bias(); }
+  [[nodiscard]] bool is_loaded() const { return is_loaded_; }
+
+  // relative_address here is the absolute address minus the address this module was loaded at by
+  // the process (module start address)
+  [[nodiscard]] orbit_client_protos::FunctionInfo* FindFunctionByRelativeAddress(
+      uint64_t relative_address, bool is_exact) const;
+  // TODO(antonrohr): The module_start_address parameter should not be needed here, but it is
+  // because FunctionInfo still contains the field module_base_address. As soon as that field is
+  // gone, remove the parameter here
+  void LoadSymbols(const orbit_grpc_protos::ModuleSymbols& module_symbols,
+                   uint64_t module_start_address);
+  // TODO(antonrohr): Remove this method as soon as FunctionInfo does not contain an absolute
+  // address anymore
+  void ClearSymbols();
+  [[nodiscard]] orbit_client_protos::FunctionInfo* FindFunctionFromHash(uint64_t hash) const {
+    return hash_to_function_map_.contains(hash) ? hash_to_function_map_.at(hash) : nullptr;
+  }
+
+  [[nodiscard]] std::vector<const orbit_client_protos::FunctionInfo*> GetFunctionList() const;
+  [[nodiscard]] absl::flat_hash_map<uint64_t, const orbit_client_protos::FunctionInfo*>
+  GetOrbitFunctions() const;
+
+ private:
+  mutable absl::Mutex functions_mutex;
```

Comment:
I think this should protect not only functions but also access to other fields, like is_loaded.

Reply:
I agree. I renamed the mutex to a general `mutex_` and also locked it in the functions I forgot.